src/Decompiler/Typing/ExpressionTypeDescender.cs

```diff
@@ -216,7 +218,19 @@ public bool VisitArrayAccess(ArrayAccess acc, TypeVariable tv)
             TypeVariable tvField)
         {
             var dtElement = factory.CreateStructureType(null, elementSize);
-            dtElement.Fields.Add(0, tvField);
+            if (
+                tvField.DataType is StructureType strField &&
+                strField.Size == 0)
```

Comment:
LEt's add a comment explaining why this branch is not taken if `strField.Size != 0`

Reply:
Done in faacfdf.